fields/types/cloudinaryimage/CloudinaryImageField.js

```diff
@@ -214,10 +260,85 @@ module.exports = Field.create({
 				target="__blank"
 				style={{ float: 'left', marginRight: '1em' }}
 			>
-				<img src={this.getImageSource()} style={{ height: 90 }} />
+				<img src={this.getImageSource()} style={{ height: 90, minWidth: 90 }} />
+				{renderIcons()}
 			</ImageThumbnail>
 		);
 	},
+	renderOption (option, isValue) {
+		const className = `octicon octicon-${this.getSelectIcon(option.resource_type)}`;
+		const styles = {
+			image: {
+				height: isValue ? 24 : 50,
+				width: isValue ? 48 : 100,
+				marginRight: '1em',
+			},
+			glyph: {
+				paddingRight: '0.5em',
+			},
+		};
+		let label = option.label;
+		if (option.filename) {
+			label = option.filename;
+		}
+		return (
+			<div>
+				<img alt={label} src={option.thumbnail} style={styles.image} />
+				<span><i style={styles.glyph} className={className} />{label}</span>
+			</div>
+		);
+	},
+	handleSelectOnChange (option) {
+		this.setState({ selectValue: option ? option.value : null });
+	},
+	handleLoadOptions (input, callback) {
+		const { listPath, path, selectPrefix } = this.props;
+
+		let prefix = '';
+		if (selectPrefix) {
+			prefix = selectPrefix;
+		} else if (Keystone.options.cloudinaryFolders) {
+			prefix = Keystone.options.cloudinaryPrefix ? `${Keystone.options.cloudinaryPrefix}/` : '';
+			prefix += `${listPath}/${path}`;
+		}
+
+		$.get('/keystone/api/cloudinary/autocomplete', {
```

Comment:
Works pretty well, I did see that this line needs updating to this.
`${Keystone.options.cloudinaryPrefix}/api/cloudinary/autocomplete`